api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -228,4 +228,13 @@ default AppendFiles newFastAppend() {
    * @return a {@link LocationProvider} to provide locations for new data files
    */
   LocationProvider locationProvider();
+
+  /**
+   * Return the name of this table.
+   *
+   * @return this table's name
+   */
+  default String name() {
```

Comment:
This newly added method should be well defined, should it return `TableIdentifier` or just `String`?